cmd/kubectl-gadget/version.go

```diff
@@ -33,32 +46,70 @@ var versionCmd = &cobra.Command{
 	Short:        "Show version",
 	SilenceUsage: true,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		fmt.Printf("Client version: v%s\n", version.Version())
+		outputFormat, err := cmd.Flags().GetString("output")
+		if err != nil {
+			return fmt.Errorf("error getting output format: %w", err)
+		}
 
+		// Initialize version info structure
+		versionInfo := &VersionInfo{
+			ClientVersion: &Version{
+				Version: version.Version().String(),
+			},
+		}
+
+		// Get server version information
 		gadgetNamespaces, err := utils.GetRunningGadgetNamespaces()
 		if err != nil {
 			return fmt.Errorf("searching for running Inspektor Gadget instances: %w", err)
 		}
 
 		switch len(gadgetNamespaces) {
 		case 0:
-			fmt.Println("Server version:", "not installed")
-			return nil
+			versionInfo.ServerVersion = &Version{
+				Status: "not installed",
+			}
 		case 1:
 			// Exactly one running gadget instance found, use it
 			runtimeGlobalParams.Set(grpcruntime.ParamGadgetNamespace, gadgetNamespaces[0])
+			info, err := grpcRuntime.InitDeployInfo()
+			if err != nil {
+				return fmt.Errorf("loading deploy info: %w", err)
+			}
+			versionInfo.ServerVersion = &Version{
+				Version: info.ServerVersion,
+			}
 		default:
-			// Multiple running gadget instances found, error out
 			return fmt.Errorf("multiple running Inspektor Gadget instances found in following namespaces: %v", gadgetNamespaces)
 		}
 
-		info, err := grpcRuntime.InitDeployInfo()
-		if err != nil {
-			return fmt.Errorf("loading deploy info: %w", err)
+		// Output based on format
+		switch outputFormat {
+		case "json":
+			return outputJSON(versionInfo)
+		case "":
+			return outputDefault(versionInfo)
+		default:
+			return fmt.Errorf("invalid output format: %s", outputFormat)
 		}
+	},
+}
 
-		fmt.Printf("Server version: v%s\n", info.ServerVersion)
+func outputJSON(info *VersionInfo) error {
+	output, err := json.MarshalIndent(info, "", "  ")
+	if err != nil {
+		return fmt.Errorf("error marshaling version info: %w", err)
+	}
+	fmt.Println(string(output))
+	return nil
+}
 
-		return nil
-	},
+func outputDefault(info *VersionInfo) error {
+	fmt.Printf("Client version: v%s\n", info.ClientVersion.Version)
+	if info.ServerVersion.Status != "" {
+		fmt.Printf("Server version: %s\n", info.ServerVersion.Status)
+	} else {
+		fmt.Printf("Server version: v%s\n", info.ServerVersion.Version)
+	}
```

Comment:
What about the following logic:

If server version is not empty, print it. Otherwise print a generic "not-available" message when it's empty?

Reply:
yes we can do it